CMakeLists.txt

```diff
@@ -47,7 +47,7 @@ if(HAS_LIBDISPATCH_API)
   find_package(dispatch CONFIG REQUIRED)
 endif()
 
-find_package(ICU COMPONENTS uc i18n REQUIRED)
+find_package(ICU COMPONENTS uc i18n REQUIRED OPTIONAL_COMPONENTS data)
```

Comment:
We need to add data to be found for sure, but can you note here the cases where it'll be optional?

Reply:
I think this is a question for @compnerd (as I just cherry-picked his commit).

Reply:
data is optional as we may be dynamically loading the data, in which case, the application does not link against the data, ICU will internally link and load the data as necessary.